src/algorithms/util/matrixToListConverter.java

```diff
@@ -0,0 +1,35 @@
+package src.algorithms.util;
+import java.util.*;
+/**
+ * Supports method to convert an Adjacency Matrix into an Adjacency list
+ * Adjacency Matrix in the form of a 2D Integer Array int[][]
+ * Adjacency list in the form of a 2D Integer array int[][], where array[i] is a list of neighbour indexes
+ */
+
+public class matrixToListConverter {
+
+    public int[][] convert(int[][] adjM) {
+
+        List<ArrayList<Integer>> adjL = new ArrayList<>();
+        int numOfNodes = adjM.length;
+
+        for (int i = 0; i < numOfNodes; i++) {
+            adjL.add(new ArrayList<>());
+            for (int j = 0; j < numOfNodes; j++) {
+                if (adjM[i][j] == 1) {
```

Comment:
Im sure your code does, but do double check if this will work on both directed/undirected graphs.

Also, add some test code for this sub-routine function too! It will likely be used in many places, so we should make sure its free of bugs.